scripts/generate-open-api-client.js

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env node
+
+const arg = require('arg');
+const { exec } = require('child_process');
+
+const { log, error } = console;
+
+const readArgs = () => {
+	const args = arg(
+		{
+			'--help': Boolean,
+			'-h': '--help',
+
+			'--url': String,
+			'-u': '--url',
+
+			'--path': String,
+			'-p': '--path',
+		},
+		{
+			argv: process.argv.slice(2),
+		}
+	);
+
+	if ('--help' in args) {
+		log(`Usage: node generate-client.js [opts]
+	OPTIONS:
+		--help (-h)		Show this help.
+		--path (-p)		Path to the newly created client's directory.
+		--url (-u)		URL/path to the spec file in yml/json format.
+	`);
+		process.exit(0);
+	}
+
+	return args;
+};
+
+const extractArg = (args, shortArg, longArg) => {
+	const index = args._.findIndex((a) => a === shortArg || a === longArg);
+	if (index === -1) {
+		throw new Error(`The parameter ${longArg} or  ${longArg} is required.`);
+	}
+
+	const value = args._[index + 1];
+
+	return value;
+};
+
+const extractConfig = (args) => {
+	const url = extractArg(args, '-u', '--url');
+	const path = extractArg(args, '-p', '--path');
+
+	const config = {
+		/** url to load the open-api definition from */
+		url,
+		/** folder to save the open-api client */
+		path,
+	};
+
+	log('Create config from arguments', config, args);
+
+	return config;
+};
+
+const errorMessageContains = (includedString, error) =>
+	error && error.message && typeof error.message === 'string' && error.message.includes(includedString);
+
+const getOpenApiCommand = (configuration) => {
+	const { url, path } = configuration;
+	const command = `openapi-generator-cli generate -i ${url} -g typescript-axios -o ${path} --additional-properties=npmName=restClient,supportsES6=true,withInterfaces=true --skip-validate-spec`;
```

Comment:
Just asking out of interest: There is also an option to build it for NestJS instead of Axios (https://github.com/OpenAPITools/openapi-generator#overview). Why did you choose Axios?

Reply:
I look into it.

Reply:
I try 
const command = `openapi-generator-cli generate -i ${url} -g typescript-nestjs -o ${path} --skip-validate-spec`;
We can a lot cleaner structured modul than can be used over import { FileService } from '../filesStorageApi/v3/api/file.service'; 

The code is really really better understandable. Enforce the use of authorisation header and so on. But it use HttpService Module from '@nestjs/common'; that is deprecated. :/

I think the interpreter must be updated. Maybe it is done with the new release Version 7.x

Reply:
We should create a ticket to track this.